Example-Crypto-REXX/RXCRYPT1.rexx

```diff
@@ -0,0 +1,2431 @@
+      do
+        errMsg = 'Failed to obtain status for LPAR entry ('||i||')'
+        return fatalError(errMsg)
+      end
+ end /* endloop thru the JSON LPARs array */
+
+return 0
+
+/*******************************************************/
+/* Function:  QueryCrypto                              */
+/*                                                     */
+/* Retrieve attributes identified by                   */
+/* LPARAttribute stem variable                         */
+/*                                                     */
+/* Return 0 if successful, otherwise return -1         */
+/*******************************************************/
+QueryCrypto:
+/* Now use the retrieved CPC uri and LPAR name
+   to retrieve crypto information for the LPAR
+   We're assuming the image activation profile name 
+   matches the LPAR name
+
+   GET <CPC uri>/image-activation-profiles/<profile name>?properties
+*/
+
+If LPARAttribute.0 < 1 then
+  do
+    return fatalError('** no LPAR attributes to query **')
+  end
+
+CryptoQueryUri = CPCuri||'/image-activation-profiles/'
+CryptoQueryUri = CryptoQueryUri||LPARname
+CryptoQueryUri = CryptoQueryUri||'?properties='
+ do i = 1 to LPARAttribute.0
+   CryptoQueryUri = CryptoQueryUri||LPARAttribute.i
+
+   /* more properties still to add so need a ',' */
+   if i < LPARAttribute.0 then
+     CryptoQueryUri = CryptoQueryUri||','
+ end
+ CryptoQueryUri = CryptoQueryUri||'&cached-acceptable=true'
+
+say '========================================================='
+say 'Starting to process content for LPAR '||LPARName
+
+CryptoQueryResponse = GetRequest(CryptoQueryUri, CPCtargetName)
+
+if CryptoQueryResponse = '' Then
+  do
+    return fatalError('** Failed to retrieve Crypto info **')
+  end
+
+/* Parse the response to obtain the values of the properties queried */
+
+outLC = outLC + 1
+REXXWRT.outLC = ','||LPARName||','||LPARstatus
+
+call JSON_parseJson CryptoQueryResponse
+
+do i = 1 to LPARAttribute.0
+    if LPARAttribute.i = CRYPTOAUTH then
+      do /* simple property */
+        LPARAttributeResponse = JSON_findValue2(0, LPARAttribute.i)
+        say LPARAttribute.i||':('||LPARAttributeResponse||')'
+        outLC = outLC + 1
+        REXXWRT.outLC = ',,,'||CRYPTOAUTH||','||LPARAttributeResponse
+      end /* simple property */
+    else /* complex property with nested content */
+      do
+        ArrayResponse = JSON_findValue(0,LPARAttribute.i,HWTJ_ARRAY_TYPE)
+        if LPARAttribute.i = ASSIGNED_DOMAINS_PROP then
+          do
+            if getassignedDomains(ArrayResponse) <> 0 then
+              return fataError('failed to parse assigned domains')
+          end
+        else if LPARAttribute.i = ASSIGNED_CRYPTOS_PROP then
+          do
+            if getassignedCryptos(ArrayResponse) <> 0 then
+              return fataError('failed to parse assigned cryptos')
+          end
+        else
+          say '** unable to handle this complex property'
```

Comment:
is this an error case then?

Reply:
yes it is, changed it to return fatalError, also fixed typo in 3 places where I had fataErrror instead of fatalError (missing the 'l')

Reply:
that's a good catch! REXX definitely wouldn't have complained until it hit that spot and tried to execute it! (totally been there)